server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -59,6 +59,7 @@ public UpdateHelper(ScriptService scriptService) {
      * Prepares an update request by converting it into an index or delete request or an update response (no action).
      */
     public Result prepare(UpdateRequest request, IndexShard indexShard, LongSupplier nowInMillis) throws IOException {
+        // TODO: Don't hard-code gFields
```

Comment:
What does that mean?

Reply:
The default value for `gFields` is set to `new String[] { RoutingFieldMapper.NAME }` in two places: `TransportShardBulkAction#getStoredFieldsSpec` and `UpdateHelper#prepare`. This is a note to remind myself to refactor this.

I think we can clean this up by defining `gFields` in `ShardGetService#prepare`. We have access to `indexSettings` there, so we can determine the index version there and decide appropriately.